lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3209,21 +3241,97 @@ LogicalResult ConvertAtenOp<AtenGeluOp>::matchAndRewrite(
         op, "Only floating-point datatype legalization supported");
   }
 
-  // TODO: Handle approximate.
+  auto resultType =
+      dyn_cast<TensorType>(getTypeConverter()->convertType(op.getType()));
+
   std::string approximate;
-  if (!matchPattern(op.getApproximate(), m_TorchConstantStr(approximate)) ||
-      approximate != "none") {
-    return rewriter.notifyMatchFailure(op, "Unsupported value of approximate");
+  if (!matchPattern(op.getApproximate(), m_TorchConstantStr(approximate))) {
+    return rewriter.notifyMatchFailure(
+        op, "Non-const approximate value not supported");
   }
 
-  Value cdf = buildUnitNormalCdf(rewriter, op, adaptor.getSelf(), selfElemTy);
-  cdf = rewriter.createOrFold<tosa::CastOp>(
-      op->getLoc(),
-      cast<RankedTensorType>(cdf.getType()).cloneWith({}, selfElemTy), cdf);
+  if (approximate.compare("none") == 0) {
+    // GELU(x) = x * CDF(x)
+    Value cdf = buildUnitNormalCdf(rewriter, op, adaptor.getSelf(), selfElemTy);
+    cdf = rewriter.createOrFold<tosa::CastOp>(
+        op->getLoc(),
+        cast<RankedTensorType>(cdf.getType()).cloneWith({}, selfElemTy), cdf);
+
+    rewriter.replaceOpWithNewOp<tosa::MulOp>(op, resultType, self, cdf,
+                                             /*shift=*/0);
+  } else if (approximate.compare("tanh") == 0) {
+    // "tanh" approximate
+    // GELU(x) = 0.5 * x * (1 + Tanh(sqrt(2/pi) * (x + 0.044715 * x^3))
+    auto selfShape = selfType.getShape();
+    auto numElem = std::accumulate(selfShape.begin(), selfShape.end(), 1,
```

Comment:
I've added a static shape check for tanh approximate. I've tried using size 1 constant but for some reasons it crashed fx_importer_target (specifically the `ElementwiseGeluApproximateTanhModule_basic` test), so I will resort to static shape requirement for now since TOSA currently only supports static shape.